### PR TITLE
Adding milter-clamav

### LINKED ISSUE #22: clamav-milter
Hi,

Would it be possible to add clamav-milter support to this module?  I personnaly use clamav mostly for mail servers and this module only lacks clamav-support to be able to add virus scanning to postfix or sendmail.

### PR DESCRIPTION
**Pull request by ubellavance**
Adding milter-clamav, take 2

Closes #22

### CODE REVIEW

**Comment by ubellavance** [comment]
Hi @edestecd, 

I think this PR is OK but Travis checks fail because I made the module only compatible with RHEL7 and the likes. Can you please have a look at it?

**Comment by edestecd** [comment]
The travis failures are unrelated to your changes. I'll get them fixed and have a look.
Thanks!

**Comment by ubellavance** [comment]
Super, thanks! Should we close #26 since I've deleted the repo and started again with this PR?

**Comment by edestecd** [comment]
Yes, go ahead and close the old PR

File: manifests/clamav_milter.pp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  unless ($::osfamily == 'RedHat') and (versioncmp($::operatingsystemrelease, '7.0') >= 0) {
+    fail("OS family ${::osfamily}-${::operatingsystemrelease} is not supported. Only RedHat >= 7 is suppported.")
+  }
+
+  $config_options = $clamav::_clamav_milter_options
+
+  package { 'clamav_milter':
+    ensure => $clamav::clamav_milter_version,
+    name   => $clamav::clamav_milter_package,
+    before => File['clamd.conf'],
```
Comment (edestecd) [inline manifests/clamav_milter.pp]: Should be:
```puppet
before => File['clamav-milter.conf'],
```

**Review by edestecd — COMMENTED** [review]
Everything looks good, except the one typo I mentioned.   

You might add an example to usage, in the README.md, so others will know how to use this.  

We also need a few spec tests, but I can help with that...

**Comment by edestecd** [comment]
@ubellavance great work! Thanks. I'm excited to add this feature to the module...

File: README.md [hunk 2]
```diff
@@ -107,6 +109,30 @@ class { 'clamav':
 }
 ```
 
+###Add clamav-milter support and customize its config (RHEL7 and derivatives only)
+####Please note that as of RHEL 7.2 only the TCP socket has been tested successfully
+```
```
Comment (edestecd) [inline README.md]: Can you change this to:
\```puppet

So the syntax highlighting will work

File: README.md [hunk 3]
```diff
@@ -107,6 +109,30 @@ class { 'clamav':
...
+  class { 'clamav':
+  manage_repo              => false,
```
Comment (edestecd) [inline README.md:117]: Can you fix the indentation on all the params...

File: README.md [hunk 4]
```diff
@@ -107,6 +109,30 @@ class { 'clamav':
...
+    'AddHeader'            => 'add',
+    'OnInfected'           => 'Reject',
+    'RejectMsg'            => 'Message rejected: Infected by %v',
+  },
+
+  manage_clamd             => true,
+  manage_freshclam         => true,
+  manage_clamav_milter     => true,
+  clamd_service_ensure     => 'running',
+  }
```
Comment (edestecd) [inline README.md:133]: I think this } is not needed.

**Review by edestecd — COMMENTED** [review]
Great documentation. IT should help anyone trying to use this. I just had a few comments that might improve it slightly.

I'll go ahead and accept the MR after you have a look.
We can do a separate MR for the spec tests, if you are up to it.

**Comment by ubellavance** [comment]
I think I did all the suggested fix.  Yes a separate PR would be OK for the spec tests. I'm not too familiar with these at all.

**Pull request merged**